All easy gfg solution added

Vipullakum007 opened the pull request:
## Related Issue

fixed : #386 

## Description

i have added solution for all easy problems
**i have added 10 files for solution of 10 problems**

![Screenshot 2024-06-05 221647](https://github.com/CodeHarborHub/codeharborhub.github.io/assets/127659443/c92f2c54-f6ea-49ff-91fe-092991a08582)

## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [ ] Documentation update
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

![Screenshot (677)](https://github.com/CodeHarborHub/codeharborhub.github.io/assets/127659443/5204ea13-0361-40a4-9c47-e3e45266e064)

![Screenshot (678)](https://github.com/CodeHarborHub/codeharborhub.github.io/assets/127659443/e468ff74-15a8-4b62-a4d9-4c350f7c1dab)

![Screenshot (679)](https://github.com/CodeHarborHub/codeharborhub.github.io/assets/127659443/31787745-7338-4616-ae76-e25df9e52ad0)

![Screenshot (680)](https://github.com/CodeHarborHub/codeharborhub.github.io/assets/127659443/f10ba663-3a48-469d-a560-c88b4cbac594)

![Screenshot (681)](https://github.com/CodeHarborHub/codeharborhub.github.io/assets/127659443/2092c690-6fc0-45cd-af86-4e9b4b254c73)

![Screenshot (682)](https://github.com/CodeHarborHub/codeharborhub.github.io/assets/127659443/ec62a601-46ff-4d85-afcf-4cee7d489152)

![Screenshot (683)](https://github.com/CodeHarborHub/codeharborhub.github.io/assets/127659443/59b90810-c174-46f5-b522-c539b2544fd8)

![Screenshot (684)](https://github.com/CodeHarborHub/codeharborhub.github.io/assets/127659443/f77acd92-7460-4892-8104-e071bf657e00)

![Screenshot (685)](https://github.com/CodeHarborHub/codeharborhub.github.io/assets/127659443/51fbd6ee-2ad0-4c0a-ac51-870846f1c90e)

![Screenshot (686)](https://github.com/CodeHarborHub/codeharborhub.github.io/assets/127659443/cf901f5d-66ec-40ff-98e2-dcb68f8f1f75)

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.
<!-- [X] - put a cross/X inside [] to check the box -->

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

Vipullakum007 commented:
hey @Ajay-Dhangar , please make it` level 3 `because i have added solution for all easy problems of gfg there are 10 problem `i have added 10 solution`

Ajay-Dhangar commented:
> hey @Ajay-Dhangar , please make it`level 3`because i have added solution for all easy problems of gfg there are 10 problem `i have added 10 solution`

@Vipullakum007 I prefer quality over quantity. I don't need multiple files or folders . You can work on a single file that I assign, possibly at level 3, but ensure it is done professionally. Please make sure your content is clear, well-structured, and thoroughly explained.

Ajay-Dhangar commented on dsa-solutions/gfg-solutions/0023-Find-triplet-with-zero-sum.md:
```diff
@@ -0,0 +1,194 @@
+
+### Constraints
+- 1 ≤ N ≤ 10^5  
+- -10^3 ≤ arr[i] ≤ 10^3
```
replace 

```md
- 1 ≤ N ≤ 10^5  
- -10^3 ≤ arr[i] ≤ 10^3
```

to 

```md
- $1 ≤ N ≤ 10^5$
- $-10^3 ≤ arr[i] ≤ 10^3$
```

Ajay-Dhangar commented on dsa-solutions/gfg-solutions/0024-minimum-indexed-character.md:
```diff
@@ -0,0 +1,180 @@
+**Expected Auxiliary Space:** O(Number of distinct characters)
+
+### Constraints
+- 1 ≤ |str|,|patt| ≤ 10^5
```
replace

```md
- 1 ≤ |str|,|patt| ≤ 10^5
- 'a' ≤ str[i], patt[i] ≤ 'z'
```

to

```md
- $1 ≤ |str|,|patt| ≤ 10^5$
- $'a' ≤ str[i], patt[i] ≤ 'z'$
```

Ajay-Dhangar commented on dsa-solutions/gfg-solutions/0024-minimum-indexed-character.md:
```diff
@@ -0,0 +1,180 @@
+
+### Complexity Analysis
+
+- **Time Complexity:** O(N), where N is the length of the string `str`. We iterate through each character in `patt` and use the `find` or `indexOf` method, which runs in O(N) time.
```
replace `O(N)` to `$O(N)$` and `O(1)` to `$O(1)$`

Ajay-Dhangar commented on dsa-solutions/gfg-solutions/Easy problems/bfs-of-graph.md:
```diff
@@ -0,0 +1,222 @@
+
+You don't need to read input or print anything. Your task is to complete the function `bfsOfGraph()` which takes the integer `V` denoting the number of vertices and adjacency list as input parameters and returns a list containing the BFS traversal of the graph starting from the 0th vertex from left to right.
+
+**Expected Time Complexity:** O(V + E)  
```
replace `O(V + E)` to `$O(V + E)$` and `O(V)` to `$O(V)$`

Ajay-Dhangar commented on dsa-solutions/gfg-solutions/Easy problems/bfs-of-graph.md:
```diff
@@ -0,0 +1,222 @@
+
+The provided solutions efficiently perform a Breadth First Search (BFS) traversal of a directed graph. By starting from the 0th vertex and using a queue to manage the traversal, the algorithms ensure that all reachable vertices are visited in the correct order. The solutions operate in O(V + E) time and use O(V) space complexity, where V and E are the numbers of vertices and edges in the graph, r ...
+
+**Time Complexity:** O(V + E)  
```
replace

`O(V + E)` $=>$ `$O(V + E)$`
`O(V)` $=>$ `$O(V)$`

Ajay-Dhangar commented on dsa-solutions/gfg-solutions/Easy problems/square-root.md:
```diff
@@ -0,0 +1,200 @@
+
+The provided solutions efficiently find the floor value of the square root of a given integer `x` using binary search. This approach ensures a time complexity of O(log N) and an auxiliary space complexity of O(1). The algorithms are designed to handle large values of `x` up to 10^7 efficiently without relying on built-in square root functions.
+
+**Time Complexity:** O(log N)  
```
replace `O(log N)` to `$O(log N)$` and `O(1)` to `$O(1)$`

Ajay-Dhangar commented on dsa-solutions/gfg-solutions/Easy problems/square-root.md:
```diff
@@ -0,0 +1,200 @@
+**Expected Auxiliary Space:** O(1)
+
+**Constraints**
+- 1 ≤ x ≤ 10^7
```
replace `1 ≤ x ≤ 10^7` to `$1 ≤ x ≤ 10^7$`

Ajay-Dhangar commented on dsa-solutions/gfg-solutions/Easy problems/square-root.md:
```diff
@@ -0,0 +1,200 @@
+
+You don't need to read input or print anything. The task is to complete the function `floorSqrt()` which takes `x` as the input parameter and returns its square root. Note: Try solving the question without using the sqrt function. The value of `x` ≥ 0.
+
+**Expected Time Complexity:** O(log N)  
```
replace` O(log N)` to `$O(log N)$` and `O(1)` to `$O(1)$`

Ajay-Dhangar commented on dsa-solutions/gfg-solutions/Easy problems/reverse-a-linked-list.md:
```diff
@@ -0,0 +1,189 @@
+### Complexity Analysis
+
+- **Time Complexity:** O(N), where N is the number of nodes in the linked list. We traverse the entire list once.
+- **Space Complexity:** O(1), as we only use a constant amount of extra space for pointers.
```
replace `O(1)` to `$O(1)$` and `O(N)` to `$O(N)$`

Ajay-Dhangar commented on dsa-solutions/gfg-solutions/Easy problems/reverse-a-linked-list.md:
```diff
@@ -0,0 +1,189 @@
+**Expected Auxiliary Space:** O(1)
+
+### Constraints
+1 ≤ N ≤ 10^4
```
**Bad practice**

```md
### Constraints
1 ≤ N ≤ 10^4
```

**Good practice**

```md
### Constraints

- $1 ≤ N ≤ 10^4$
```

Ajay-Dhangar commented on dsa-solutions/gfg-solutions/Easy problems/reverse-a-linked-list.md:
```diff
@@ -0,0 +1,189 @@
+### Your Task
+The task is to complete the function `reverseList()` with head reference as the only argument and should return the new head after reversing the list.
+
+**Expected Time Complexity:** O(N)  
```
replace `O(N)` to `$O(N)$`
and `O(1)` to `$O(1)$`

Ajay-Dhangar reviewed (changes requested):
Now update code for all files based on suggestions

Vipullakum007 commented:
> > hey @Ajay-Dhangar , please make it`level 3`because i have added solution for all easy problems of gfg there are 10 problem `i have added 10 solution`
> 
> @Vipullakum007 I prefer quality over quantity. I don't need multiple files or folders . You can work on a single file that I assign, possibly at level 3, but ensure it is done professionally. Please make sure your content is clear, well-structured, and thoroughly explained.

please can you point out what you want so that i can better / i can make professionally
i have ask this question many times please answer me.

Vipullakum007 commented:
> > > hey @Ajay-Dhangar , please make it`level 3`because i have added solution for all easy problems of gfg there are 10 problem `i have added 10 solution`
> > 
> > 
> > @Vipullakum007 I prefer quality over quantity. I don't need multiple files or folders . You can work on a single file that I assign, possibly at level 3, but ensure it is done professionally. Please make sure your content is clear, well-structured, and thoroughly explained.
> 
> please can you point out what you want so that i can better / i can make professionally i have ask this question many times please answer me.

hey @Ajay-Dhangar , updated as per your suggestion , you can review it. please answer

Ajay-Dhangar commented:
> > > hey @Ajay-Dhangar , please make it`level 3`because i have added solution for all easy problems of gfg there are 10 problem `i have added 10 solution`
> > 
> > 
> > @Vipullakum007 I prefer quality over quantity. I don't need multiple files or folders . You can work on a single file that I assign, possibly at level 3, but ensure it is done professionally. Please make sure your content is clear, well-structured, and thoroughly explained.
> 
> please can you point out what you want so that i can better / i can make professionally i have ask this question many times please answer me.

Yah sure, 
I review your work then may be assign good level if your contents is good and fully based on our structures

Ajay-Dhangar commented on dsa-solutions/gfg-solutions/Easy problems/bfs-of-graph.md:
```diff
@@ -0,0 +1,222 @@
+**Expected Auxiliary Space:** $O(V)$
+
+**Constraints**
+- $ 1 ≤ V, E ≤ 10^4 $
```
replace `- $ 1 ≤ V, E ≤ 10^4 $` to `- $1 ≤ V, E ≤ 10^4$`

Ajay-Dhangar commented on dsa-solutions/gfg-solutions/Easy problems/bfs-of-graph.md:
```diff
@@ -0,0 +1,222 @@
+
+The provided solutions efficiently perform a Breadth First Search (BFS) traversal of a directed graph. By starting from the 0th vertex and using a queue to manage the traversal, the algorithms ensure that all reachable vertices are visited in the correct order. The solutions operate in O(V + E) time and use O(V) space complexity, where V and E are the numbers of vertices and edges in the graph, r ...
+
+**Time Complexity:** $ O(V + E)  $
```
not using space

and replace `$ O(V + E) $` to `$O(V + E)$`

and also, `$ O(V)$` to `$O(V)$`

Ajay-Dhangar commented on dsa-solutions/gfg-solutions/Easy problems/check-for-balanced-tree.md:
```diff
@@ -0,0 +1,255 @@
+### Constraints
+
+- The number of nodes in the tree is in the range `[0, 5000]`.
+- $ -10^4 <=$ Node.val $<= 10^4 $
```
replace `- $ -10^4 <=$ Node.val $<= 10^4 $` to `- $-10^4 <=$ Node.val $<= 10^4$`

Ajay-Dhangar commented on dsa-solutions/gfg-solutions/Easy problems/dfs-of-graph.md:
```diff
@@ -0,0 +1,204 @@
+
+### Complexity Analysis
+
+- **Time Complexity:** $ O(V + E)  $
```
same issue

Ajay-Dhangar commented on dsa-solutions/gfg-solutions/Easy problems/implement-two-stacks-in-an-array.md:
```diff
@@ -0,0 +1,349 @@
+
+## Complexity Analysis
+
+The provided solutions efficiently implement two stacks in a single array using pointers. This approach ensures a time complexity of O(1) for all stack operations and an auxiliary space complexity of O(1). The algorithms are designed to handle up to 10^4 queries efficiently without relying on dynamic data structures.
```
replace `O(1)` to `$O(1)$`

Ajay-Dhangar reviewed (changes requested):
update your code based on suggestions and can't be use space for maths expression. And review all files code and resolve issues

Vipullakum007 commented:
hey @Ajay-Dhangar  ,updated as per your suggestion

Ajay-Dhangar reviewed (approved):
Done
